### PR TITLE
add support for threading in cmeps

### PR DESCRIPTION
**Pull request by jedwards4b**
Add support for threading in cmeps driver.

### CODE REVIEW

**Comment by alperaltuntas** [comment]
Nuopc tests are failing with the latest CESM_nuopc_dev. I will try running with the master cime, cmeps, and cdeps.

  ERS_Vnuopc.T62_g16.CMOM_IAF.cheyenne_intel (Overall: FAIL) details:
    FAIL ERS_Vnuopc.T62_g16.CMOM_IAF.cheyenne_intel RUN time=3
  ERS_Vnuopc.T62_t061.GMOM.cheyenne_intel (Overall: FAIL) details:
    FAIL ERS_Vnuopc.T62_t061.GMOM.cheyenne_intel RUN time=5
  SMS_D_Vnuopc.T62_t061.GMOM_IAF.cheyenne_intel (Overall: FAIL) details:
    FAIL SMS_D_Vnuopc.T62_t061.GMOM_IAF.cheyenne_intel RUN time=6
  SMS_Vnuopc.T62_t025.CMOM.cheyenne_intel (Overall: FAIL) details:
    FAIL SMS_Vnuopc.T62_t025.CMOM.cheyenne_intel RUN time=4

File: config_src/drivers/nuopc_cap/mom_cap.F90 [hunk 1]
```diff
@@ -435,7 +438,31 @@ subroutine InitializeAdvertise(gcomp, importState, exportState, clock, rc)
...
+          line=__LINE__, &
+          file=__FILE__)) &
+          return  ! bail out
+     read(cvalue,*) nthrds
+  else
+     nthrds = localPeCount
+  endif
+
+!$  call omp_set_num_threads(nthrds)
+    print *,__FILE__,__LINE__,nthrds
```
Comment (alperaltuntas) [inline config_src/drivers/nuopc_cap/mom_cap.F90]: These stray print statements need to be removed. There is also one in line 901

**Review by alperaltuntas — CHANGES_REQUESTED** [review]

**Comment by alperaltuntas** [comment]
After switching to master cime, cmeps, and cdeps, The nuopc tests are now passing. Note that I get NLCOMP failures but I believe it has to do with new the namelist variable drv_threading.

File: config_src/drivers/nuopc_cap/mom_cap.F90 [hunk 2]
```diff
@@ -143,7 +144,8 @@ module MOM_cap_mod
 integer              :: scalar_field_count = 0
 integer              :: scalar_field_idx_grid_nx = 0
 integer              :: scalar_field_idx_grid_ny = 0
-character(len=*),parameter :: u_FILE_u = &
+integer              :: nthrds  !< number of openmp threads per task
+character(len=*),parameter :: u_file_u = &
```
Comment (alperaltuntas) [inline config_src/drivers/nuopc_cap/mom_cap.F90]: I think, for consistency, we should change `u_file_u` back to `u_FILE_u`. (Not sure why it's changed in the first place.)

**Review by alperaltuntas — CHANGES_REQUESTED** [review]

File: config_src/drivers/nuopc_cap/mom_cap.F90 [hunk 3]
```diff
@@ -847,6 +875,32 @@ subroutine InitializeRealize(gcomp, importState, exportState, clock, rc)
...
+          line=__LINE__, &
+          file=__FILE__)) &
+          return  ! bail out
+     read(cvalue,*) nthrds
+  else
+     nthrds = localPeCount
+  endif
+
+!$  call omp_set_num_threads(nthrds)
+    print *,__FILE__,__LINE__,nthrds
```
Comment (alperaltuntas) [inline config_src/drivers/nuopc_cap/mom_cap.F90]: need to remove this print statement.

**Review by alperaltuntas — CHANGES_REQUESTED** [review]

**Comment by alperaltuntas** [comment]
Thanks, @jedwards4b . Could you also remove the print statements. I'll then merge this and create a new tag.

**Comment by jedwards4b** [comment]
I did remove the print statements - are you looking at the latest code?

**Pull request merged**